src/dfx/src/commands/extension/list.rs

```diff
@@ -0,0 +1,7 @@
+use crate::lib::environment::Environment;
+use crate::lib::error::DfxResult;
+
+pub fn exec(env: &dyn Environment) -> DfxResult<()> {
+    env.get_cache().list_installed_extensions();
```

Comment:
It might be helpful to think of these commands as calling some high-level methods, and displaying the results.

The current implementation, as expressed in `list_installed_extensions()`, combines the details of how to get the list of installed extensions with how to present the output of the comment.

The list of installed extensions may be "the names of the directories in the extensions directory", or there may be more to it.  See https://github.com/dfinity/sdk/blob/master/src/dfx/src/config/cache.rs#L301 for an example.

So something like this, in this `exec` function, similar to [dfx cache list](https://github.com/dfinity/sdk/blob/master/src/dfx/src/commands/cache/list.rs):
```
    let mgr = ExtensionsManager::new(env); // or maybe env.get_extension_manager()
    let mut names = mgr.installed_extensions()
        .map(|ext|ext.name());
    if names.is_empty() {
        println!("No extensions installed");
    } else {
        println!("Installed extensions:");
        names.sort();
        for name in names {
            println!("{}", name);
        }
    }
```